### DISCUSSION TITLE
Attributes missing from select attributes

### DISCUSSION THREAD

**Discussion by ergobill** [rappen/FetchXMLBuilder#667] [Q&A] [answered]
Latest version of Xrm Toolbox (v1.2022.2.54) and FetchXML Builder (1.2022.3.1)

simple account entity fetch   looking for Address 1 and Address 2 constituent fields. 

None showing in the Select Attributes Window.  
![image](https://user-images.githubusercontent.com/24599528/156592958-193e95c7-2032-4661-8495-bd28c8fb7c38.png)

no errors that I can find.

**Comment by rappen** [comment, marked as answer]
I've seen this today too...
Kinda stupid, but these attributes are *not* on `account` but in `customaddress` entity, joined with `accountid`=`parentid` and then the `addressnumber` for each address on account...
So you can join these.
Or you can get these "on" account, if you show these "Logical" attributes.

![image](https://user-images.githubusercontent.com/2572253/156605447-a45e7a34-518c-4eba-b90c-5c191a9235e0.png)

Default is not to show logical, since they add a bunch of other attributes.
You change this on menu *Options - Select Entities/Attributes*.

![image](https://user-images.githubusercontent.com/2572253/156606497-d8627eba-1dfe-45d9-ac48-da80950be9e1.png)

But address attributes are common to use...
This is not perfect, I know. But this is why and this says how.

**Reply by ergobill** [reply]
yes, this did fix it. thank you.   Without it even when I manually added the fields to the fetch the builder gives a bang light that they are invalid attributes.  But the addition of the Logical attributes fixes it.

much appreciated...

![image](https://user-images.githubusercontent.com/24599528/156609297-f9b47329-1354-4f8e-8c79-8124935edc91.png)

**Reply by rappen** [reply]
I agree, the warning message should not say it is "not valid", but rather "not currently showing"... or even better, no warning if the attribute is in the DB, but not showing. I will work on this.

**Comment by ryanwga** [comment]
@rappen 

Hello,
I ran into this today as well.  Is it best practice to use link-entity over logical attributes?  When I use link-entity, it doesn't provide any data for the addresses in the "Result View".  When I go the logical route, values show up. **(I just tested again, the logical attributes don't show up in the results either)**  Is MS going to retire these logical attributes or is it safe to continue using them?

Thanks!

**Reply by rappen** [reply]
Hi @ryanwga 

I suggest you to keep using logical attributes, that is not going away from MS.
I will make a better way to show the "obvious" attributes.
See it in next release!
Keep track of issue #668
